types/mongoose/index.d.ts

```diff
@@ -370,6 +370,8 @@ declare module "mongoose" {
     useNewUrlParser?: boolean;
     /** Set to false to make findOneAndUpdate() and findOneAndRemove() use native findOneAndUpdate() rather than findAndModify(). */
     useFindAndModify?: boolean;
+    /** False by default. Set to true to opt in to using the MongoDB driver's new connection management engine. You should set this option to true, except for the unlikely case that it prevents you from maintaining a stable connection. **/
+    useUnifiedTopology?: boolean;
```

Comment:
Seems like this already exists in `mongodb.MongoClientOptions`? See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/927dcd0af0ab47e790917fe9cc1860afc06640e1/types/mongodb/index.d.ts#L322 and http://mongodb.github.io/node-mongodb-native/3.4/api/MongoClient.html#.connect

Reply:
Yes, mongodb's Options has this field
but mongoose's ConnectionOptions doesn't.
then use mongoose with `useUnifiedTopology` will get error
```
error TS2769: No overload matches this call.
  Overload 1 of 3, '(uris: string, callback: (err: MongoError) => void): Promise<typeof import("mongoose")>', gave the following error.
    Argument of type '{ useNewUrlParser: boolean; authSource: string; useFindAndModify: boolean; useUnifiedTopology: boolean; }' is not assignable to parameter of type '(err: MongoError) => void'.
      Object literal may only specify known properties, and 'useNewUrlParser' does not exist in type '(err: MongoError) => void'.
  Overload 2 of 3, '(uris: string, options?: ConnectionOptions | undefined): Promise<typeof import("mongoose")>', gave the following error.
    Argument of type '{ useNewUrlParser: true; authSource: string; useFindAndModify: false; useUnifiedTopology: boolean; }' is not assignable to parameter of type 'ConnectionOptions'.
      Object literal may only specify known properties, and 'useUnifiedTopology' does not exist in type 'ConnectionOptions'.
```